backend/src/bundles/files/file.controller.ts

```diff
@@ -39,17 +34,33 @@ class FileController extends ControllerBase {
         this.fileService = fileService;
 
         this.addRoute({
-            path: FileApiPath.UPLOAD,
-            preHandler: uploadFile.single('document'),
+            path: FileApiPath.DOCUMENT,
+            preHandler: uploadFile.single(FileGroups.DOCUMENT),
+            method: 'POST',
+            handler: (options) => {
+                return this.upload(
+                    options as ApiHandlerOptions<{
+                        body: {
+                            file: MulterFile;
+                        };
+                    }>,
+                );
+            },
+        });
+
+        this.addRoute({
+            path: FileApiPath.IMAGE,
+            preHandler: uploadFile.single(FileGroups.IMAGE),
             method: 'POST',
-            handler: (options) =>
-                this.upload(
+            handler: (options) => {
+                return this.upload(
```

Comment:
I decided to make two endpoints because for document uploads i needed `prehandler: uploadFile.single('document')` while for photos i needed `prehandler: uploadFile.single('image')`.

I couldn't find a way of telling the controller that prehandler should be conditional. I will try to find a way again.

Reply:
I am trying something out.

Reply:
@MaksymGavatuk I managed to consolidate upload into one request.